chain/client/src/rocksdb_metrics.rs

```diff
@@ -0,0 +1,162 @@
+                            if !self.gauges.contains_key(key) {
+                                self.gauges.insert(
+                                    key.to_string(),
+                                    try_create_gauge_vec(
+                                        &get_prometheus_metric_name(stat_name),
+                                        stat_name,
+                                        &["quantile"],
+                                    )?,
+                                );
+                            }
+                            self.gauges
+                                .get(key)
+                                .unwrap()
+                                .with_label_values(&[&format!("{:.2}", *percentile as f64 * 0.01)])
+                                .set(*value);
+                        }
+                    }
+                }
+            }
+        }
+        Ok(())
+    }
+
+    fn set_int_value<F1, F2>(
+        &mut self,
+        key_fn: F1,
+        metric_fn: F2,
+        stat_name: &str,
+        value: i64,
+    ) -> anyhow::Result<()>
+    where
+        F1: Fn(&str) -> String,
+        F2: Fn(&str) -> String,
+    {
+        let key: &str = &key_fn(stat_name);
+        if !self.int_gauges.contains_key(key) {
+            self.int_gauges
+                .insert(key.to_string(), try_create_int_gauge(&metric_fn(stat_name), stat_name)?);
+        }
+        self.int_gauges.get(key).unwrap().set(value);
+        Ok(())
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
+pub(crate) enum StatsValue {
+    Count(i64),
+    Sum(i64),
+    Percentile(u32, f64),
+}
+
+/// Parses a string containing RocksDB statistics.
+pub(crate) fn parse_statistics(statistics: &str) -> anyhow::Result<Vec<(&str, Vec<StatsValue>)>> {
+    let mut result = vec![];
+    // Statistics are given one per line.
+    for line in statistics.split('\n') {
+        // Each line follows one of two formats:
+        // 1) <stat_name> COUNT : <value>
+        // 2) <stat_name> P50 : <value> P90 : <value> COUNT : <value> SUM : <value>
+        // Each line gets split into words and we parse statistics according to this format.
+        if let Some((stat_name, words)) = line.split_once(' ') {
+            let mut values = vec![];
+            let mut words = words.split(" : ").flat_map(|v| v.split(" "));
+            while let (Some(key), Some(val)) = (words.next(), words.next()) {
+                match key {
+                    "COUNT" => values.push(StatsValue::Count(val.parse::<i64>()?)),
+                    "SUM" => values.push(StatsValue::Sum(val.parse::<i64>()?)),
+                    p if p.starts_with("P") => values.push(StatsValue::Percentile(
+                        key[1..].parse::<u32>()?,
+                        val.parse::<f64>()?,
+                    )),
+                    _ => {
+                        return Err(anyhow::anyhow!(
+                            "Unsupported stats value: {} in {}",
```

Comment:
```suggestion
                            "Unsupported stats value: {key} in {line}",
```

Or is it the case that anyhow doesn’t yet supports this syntax?